tfhe/docs/getting_started/operations.md

```diff
@@ -1,64 +1,484 @@
+| Lower or equal than   | `le`        | :heavy_check_mark: | :heavy_check_mark:       | :heavy_check_mark:       |
+| Equal                 | `eq`        | :heavy_check_mark: | :heavy_check_mark:       | :heavy_check_mark:       |
+| Cast                  | `cast_into` | :heavy_check_mark: | :heavy_multiplication_x: | :heavy_multiplication_x: |
 
-The list of supported operations by the homomorphic Booleans is:
 
-| Operation Name | type    |
-| -------------- | ------- |
-| `not`          | Unary   |
-| `and`          | Binary  |
-| `or`           | Binary  |
-| `xor`          | Binary  |
-| `nor`          | Binary  |
-| `xnor`         | Binary  |
-| `cmux`         | Ternary |
+# Operations
 
-A walk-through using homomorphic Booleans can be found [here](../Boolean/tutorial.md).
+The structure and operations related to all types (i.e., Booleans, shortint and integer) are described in this section.
 
-## Shortint
+## Booleans
 
-In TFHE-rs, shortint represents short unsigned integers encoded over a maximum of 8 bits. A complete homomorphic arithmetic is provided, along with the possibility to compute univariate and bi-variate functions. Some operations are only available for integers up to 4 bits. More technical details can be found [here](../shortint/operations.md).
+Native homomorphic Booleans support common Boolean operations.
 
 The list of supported operations is:
 
-| Operation name            | Type         |
-| ------------------------- | ------------ |
-| Negation                  | Unary        |
-| Addition                  | Binary       |
-| Subtraction               | Binary       |
-| Multiplication            | Binary       |
-| Division\*                | Binary       |
-| Modular reduction         | Binary       |
-| Comparisons               | Binary       |
-| Left/Right Shift          | Binary       |
-| And                       | Binary       |
-| Or                        | Binary       |
-| Xor                       | Binary       |
-| Exact Function Evaluation | Unary/Binary |
-
-{% hint style="info" %}
-The division operation implements a subtlety: since data is encrypted, it might be possible to compute a division by 0. The division is tweaked so that dividing by 0 returns 0.
-{% endhint %}
-
-A walk-through example can be found [here](../shortint/tutorial.md), and more examples and explanations can be found [here](../shortint/operations.md).
+| name                                                          | symbol | type   |
+| ------------------------------------------------------------- | ------ | ------ |
+| [BitAnd](https://doc.rust-lang.org/std/ops/trait.BitAnd.html) | `&`    | Binary |
+| [BitOr](https://doc.rust-lang.org/std/ops/trait.BitOr.html)   | `\|`   | Binary |
+| [BitXor](https://doc.rust-lang.org/std/ops/trait.BitXor.html) | `^`    | Binary |
+| [Neg](https://doc.rust-lang.org/std/ops/trait.Neg.html)       | `!`    | Unary  |
```

Comment:
CMux maybe ? Even if it does not have its own trait in the language

Reply:
We don't have a CMUX function